pkg/controller/sdk/status_accessor_test.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sdk
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/knative/eventing-sources/pkg/apis/sources/v1alpha1"
+	duckv1alpha1 "github.com/knative/pkg/apis/duck/v1alpha1"
+)
+
+func TestNewReflectedStatusAccessor(t *testing.T) {
+	testCases := map[string]struct {
+		obj         interface{}
+		expectedErr bool
+		initial     interface{}
+		update      interface{}
+	}{
+		"obj is a string": {
+			obj:         "hello",
+			expectedErr: true,
+		},
+		"status is not present": {
+			obj:         struct{}{},
+			expectedErr: true,
+		},
+		"status is not an interface": {
+			obj: struct {
+				Status int
+			}{},
+			expectedErr: true,
+		},
+		"no status set": {
+			obj:     &v1alpha1.ContainerSource{},
+			initial: v1alpha1.ContainerSourceStatus{},
+		},
+		"set status": {
+			obj:     &v1alpha1.ContainerSource{},
+			initial: v1alpha1.ContainerSourceStatus{},
+			update: v1alpha1.ContainerSourceStatus{
+				SinkURI: "just testing",
+			},
+		},
+		"set status with conditions": {
```

Comment:
What is the difference between this test and "set status"?

Reply:
this one has conditions

Reply:
:D